modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -181,6 +188,58 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
   ): F[Either[ResolutionError, Json]] =
     lookupSchemaResult(schemaKey).map(_.map(_.value.schema))
 
+  def lookupSchemasUntil(
+    maxSchemaKey: SchemaKey
+  )(implicit
+    F: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): F[Either[SchemaResolutionError, List[RawSchema]]] = {
+
+    // If Iglu Central or any of its mirrors doesn't have a schema,
+    // it should be considered NotFound, even if one of them returned an error
+    def isNotFound(error: ResolutionError): Boolean = {
+      val (igluCentral, custom) = error.value.partition { case (repo, _) =>
+        allIgluCentral.contains(repo)
+      }
+      custom.values.flatMap(_.errors).forall(_ == RegistryError.NotFound) &&
```

Comment:
This line will always return true if `custom` is an empty list.

Reply:
:facepalm: Good spot !